src/platforms/PolygonId/components/NoDID.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  FormControl,
+  FormLabel,
+  Input,
+  ModalBody,
+  ModalHeader,
+  Spacer,
+} from "@chakra-ui/react"
+import useUser from "components/[guild]/hooks/useUser"
+import Button from "components/common/Button"
+import FormErrorMessage from "components/common/FormErrorMessage"
+import { useSubmitWithSign } from "hooks/useSubmit"
+import { Stack } from "phosphor-react"
+import { useForm, useWatch } from "react-hook-form"
+import { mutate } from "swr"
+import fetcher from "utils/fetcher"
+
+const NoDID = () => {
+  const { id } = useUser()
+  const connectPolygonId = useSubmitWithSign(
+    () => fetcher("/v1/polygon-id/connect", { method: "POST" }),
+    {
+      onSuccess: () => {
+        mutate(`/v1/users/${id}/polygon-id`)
+      },
+    }
+  )
+  const {
+    register,
+    formState: { errors },
+    control,
+    handleSubmit,
+  } = useForm({ defaultValues: { did: "" } })
+
+  const DID = useWatch({ name: "did", control })
+
+  return (
+    <form
+      onSubmit={handleSubmit(({ did }) =>
```

Comment:
We can remove this `<form>` wrapper, and just add the `onSubmit` handler on the button below